From 43ae415034763c98a28f90ce3ba1dd9a6e6f3817 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mika=20Bostr=C3=B6m?= <bostik@odin.bostik.iki.fi>
Date: Thu, 10 May 2012 11:45:33 +0300
Subject: [PATCH 1/5] Fix graphics context build

GraphicsContext3DQt.cpp references definitions which are defined only in
QtOpenGL private headers. Redefine for local use, as the private header
can not be included here.

Fix few remaining typos/artifacts where code uses wrong variable types.
---
 .../platform/graphics/qt/GraphicsContext3DQt.cpp       |   16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/Source/WebCore/platform/graphics/qt/GraphicsContext3DQt.cpp b/Source/WebCore/platform/graphics/qt/GraphicsContext3DQt.cpp
index 7005785..2e00219 100644
--- a/Source/WebCore/platform/graphics/qt/GraphicsContext3DQt.cpp
+++ b/Source/WebCore/platform/graphics/qt/GraphicsContext3DQt.cpp
@@ -43,6 +43,18 @@
 #include <texmap/TextureMapperGL.h>
 #endif
 
+/* Define GL_*_FRAMEBUFFER_EXT magic numbers. They would be available in
+ * QtOpenGL/private/qglextensions_p.h but that header can not be
+ * included here.
+ */
+#ifndef GL_READ_FRAMEBUFFER_EXT
+#define GL_READ_FRAMEBUFFER_EXT 0x8CA8
+#endif
+#ifndef GL_DRAW_FRAMEBUFFER_EXT
+#define GL_DRAW_FRAMEBUFFER_EXT 0x8CA9
+#endif
+
+
 #if ENABLE(WEBGL)
 
 namespace WebCore {
@@ -1087,7 +1099,7 @@ void GraphicsContext3D::uniform1f(GC3Dint location, GC3Dfloat v0)
     glUniform1f(location, v0);
 }
 
-void GraphicsContext3D::uniform1fv(GC3Dint location, GGC3Dsizei size, C3Dfloat* array)
+void GraphicsContext3D::uniform1fv(GC3Dint location, GC3Dsizei size, GC3Dfloat* array)
 {
     makeContextCurrent();
     glUniform1fv(location, size, array);
@@ -1135,7 +1147,7 @@ void GraphicsContext3D::uniform1i(GC3Dint location, GC3Dint v0)
     glUniform1i(location, v0);
 }
 
-void GraphicsContext3D::uniform1iv(GC3Dint location, GGC3Dsizei size, C3Dint* array)
+void GraphicsContext3D::uniform1iv(GC3Dint location, GC3Dsizei size, GC3Dint* array)
 {
     makeContextCurrent();
     glUniform1iv(location, size, array);
-- 
1.7.10

